options.go

```diff
@@ -77,6 +84,19 @@ func checkOptions(rp *Ringpop) []error {
 
 // Runtime options
 
+// Clock is used to set the Clock mechanism. Passing nil will use the system
+// clock. Testing harnesses will typically replace this with a mocked clock.
+func Clock(c clock.Clock) Option {
+	return func(r *Ringpop) error {
+		if c == nil {
```

Comment:
Does this mean that we now have two ways to set a default value. I think calling it with nil should cause a panic or return an error.